model/activity_resampling.py

```diff
@@ -121,6 +121,31 @@ def health_status_adjustment(self):
 
         return self._update(self.hsa.run(self.run_params))
 
+    def inequalities_adjustment(self):
+        """perform the inequalities adjustment"""
+        activity_type = self._activity_type
+        if activity_type == "aae":
+            return self
+
+        if not (params := self.run_params["inequalities"]):
+            return self
+
+        factor = pd.concat(
+            {
+                k: pd.Series(v, name="inequalities", dtype="float64")
+                for k, v in params.items()
+            }
+        )
+        factor.index.names = ("sushrg_trimmed", "imd_quintile")
+        factor.index = factor.index.set_levels(
+            factor.index.levels[1].astype(int), level="imd_quintile"
+        )
+        if activity_type == "op":
+            factor = pd.concat({"procedure": factor}, names=["group"])
+        if activity_type == "ip":
+            factor = pd.concat({"elective": factor}, names=["group"])
```

Comment:
I'm not a big fan on this code block; it's not very DRY, and the if statements should be collapsed.

```suggestion
match activity_type:
  case "op":
    factor_key = "procedure"
  case "ip":
    factor_key = "elective"
    
factor = pd.concat({factor_key: factor}, names=["group"])
```

the behaviour of this coded if we have some different activity type ends up with `factor_key` being undefined, so the last line will error.

this leads me to think what we should do is ensure on line 127 activity_type is either "ip" or "op", and then `return self` if it isn't.

you could then do something simpler like `factor_key = "procedure" if activity_type == "op" else "elective"`